ezyrb/nnspod.py

```diff
@@ -0,0 +1,269 @@
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+import numpy as np
+import torch
+import torch.nn as nn
+from .ann import ANN
+from .pod import POD
+from .database import Database
+
+
+class NNsPOD(POD):
+    def __init__(self, method = "svd", path = None):
+        super().__init__(method)
+        self.path = path
+    
+
+    def reshape2dto1d(self, x, y):
+        """
+        reshapes graphable data into data that can go through the Neural Net
+        :param array x: x value of data
+        :param array y: y value of data
+        """
+        x = x.reshape(-1,1)
+        y = y.reshape(-1,1)
+        coords = np.concatenate((x, y), axis = 1)
+        coords = np.array(coords).reshape(-1,2)
+        
+        
+        return coords
+
+    def reshape1dto2d(self, snapshots):
+        """
+        turns 1d list of data into 2d
+        :param array-like snapshots: data to be reshaped
+        """
+        return snapshots.reshape(int(np.sqrt(len(snapshots))), int(np.sqrt(len(snapshots))))
+
+
+    def train_InterpNet1d(self,ref_data, interp_layers, interp_function, interp_stop_training, interp_loss, retrain = False):
+        """
+        trains the Interpnet given 1d data:
+
+        :param database ref_data: the reference data that the rest of the data will be shifted to
+        :param list interp_layers: list with number of neurons in each layer
+        :param torch.nn.modules.activation interp_function: activation function for the interpnet
+        :param float interp_stop_training: desired tolerance for the interp training
+        :param torch.nn.Module interp_loss: loss function (MSE default)
+        :param boolean retrain: True if the interpNetShould be retrained, False if it should be loaded
+        """
+        
+        self.interp_net = ANN(interp_layers, interp_function, interp_stop_training, interp_loss)
+        if not retrain:
+            try:
+                self.interp_net = self.interp_net.load_state(self.path, ref_data.space.reshape(-1,1), ref_data.snapshots.reshape(-1,1))
+                print("loaded interpnet")
+            except:
+                self.interp_net.fit(ref_data.space.reshape(-1,1), ref_data.snapshots.reshape(-1,1))
+                self.interp_net.save_state(self.path)
+        else:
+            self.interp_net.fit(ref_data.space.reshape(-1,1), ref_data.snapshots.reshape(-1,1))
+            self.interp_net.save_state(self.path)
+        
+
+
+    def train_InterpNet2d(self,ref_data, interp_layers, interp_function, interp_stop_training, interp_loss, retrain = False):
```

Comment:
Please merge function for 1D e 2D. Keep always one general function, instead of specifying several methods for different dimensions